docs/src/modules/components/ThemeContext.js

```diff
@@ -97,6 +98,13 @@ if (process.env.NODE_ENV !== 'production') {
 
 const useEnhancedEffect = typeof window === 'undefined' ? React.useEffect : React.useLayoutEffect;
 
+let createMuiTheme;
+if (process.env.REACT_MODE === 'legacy') {
+  createMuiTheme = createLegacyModeTheme;
+} else {
+  createMuiTheme = createStrictModeTheme;
```

Comment:
Just curious, what is breaking about the strict mode theme? And is the default going to be strict mode, and to get the current behaviour, you'd need to explicitly set `REACT_MODE` to `legacy`?

Reply:
Children of these transition components need to be able to hold a ref.

> And is the default going to be strict mode, and to get the current behaviour, you'd need to explicitly set REACT_MODE to legacy?

You mean in user code or our code? It's not clear yet how this is going to be applied in user land. We might not document it at all if it does not fix all strict mode issues. For now this reduces warnings in our docs which helps debugging concurrent mode related issues.

Reply:
Ah gotcha!

> Children of these transition components need to be able to hold a ref.

What about adding the `findDOMNode` workaround that can be seen in other components?

Reply:
> What about adding the findDOMNode workaround that can be seen in other components? It shouldn’t fire a warning if children forward refs, right?

Yeah. If you call findDOMNode() on an HTML element it will just return that element. Only for class components you'll get the warning. We a test for that case upstreamed (https://github.com/facebook/react/pull/15091) but so far it has been ignored. I'll probably close this soon to cleanup. Now that react publishes canaries we test this approach regularly anyway.

Reply:
Sorry, in case I wasn't clear, I meant using this `findDOMNode` approach with `@material-ui/react-transition-group` (or maybe even inside the fork itself) in order for this change not to be breaking 🙂